### PR TITLE
HW1-HW4 Olesia

### PR DESCRIPTION
**Pull request by OlesiaBarannik**

### CODE REVIEW

File: OlesiaBarannik/HW1/1.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# WRITE A PROGRAM THAT WILL DISPLAY THE FOLLOWING QUESTIONS FOR USER:
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW1/1.py:1]: good job

File: OlesiaBarannik/HW1/2.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# WRITE A PROGRAM THAT WILL CALCULATE THE SUM,
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW1/2.py:1]: good job

File: OlesiaBarannik/HW3/1.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# You need to write Python's philosophy in the string format:
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW3/1.py:1]: good job
look at another implementation
print("In 'Zen of Python' the word \"better\" repeat", zen_Of_Python.count("better"), 'times')
print("In 'Zen of Python' the word \"never\" repeat", zen_Of_Python.count("never"), 'times')
print("In 'Zen of Python' the word \"is\" repeat", zen_Of_Python.count("is"), 'times')


print(zen_Of_Python.upper()) 



print(zen_Of_Python.replace("i","&"))

File: OlesiaBarannik/HW3/2.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#  A four-digit natural number is specified:
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW3/2.py:1]: good job

File: OlesiaBarannik/HW3/3.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Interchange the values of two variables without using the third variable.
+
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW3/3.py:2]: good job

File: OlesiaBarannik/HW4/1.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW4/1.py:1]: excellent work

File: OlesiaBarannik/HW3/1.py (hunk 3 above)
Comment (OlesiaBarannik) [reply OlesiaBarannik/HW3/1.py:1]: I test it, thank you

File: OlesiaBarannik/HW5/1.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Task1. Create a list that contains elements of integer type, then
+# use the loop to change the type of these elements to a floating
+# type.
+# (Hint: use the built-in float () function).
+
+
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW5/1.py:6]: good job

File: OlesiaBarannik/HW5/2.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# Task2. Print Fibonacci numbers up to the entered number n,
+# using cycles.
+# (Sequence of Fibonacci numbers 0, 1, 1, 2, 3, 5, 8, 13, etc.)
+
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW5/2.py:4]: good job
look at this implementation


def logic_of_Fibonacci(user_number):
    list_Fibonacci = [0, 1]
    if user_number == 0:
        return [0]
    elif user_number == 1:
        return list_Fibonacci

    previous_number, second_number = 0, 1
    while True:
        result = previous_number + second_number
        if result > user_number:
            break
        list_Fibonacci.append(result)
        previous_number, second_number = second_number, result

    return list_Fibonacci


user_number = int(input("Please enter a number to get its Fibonacci list\n"))
print(logic_of_Fibonacci(user_number))

File: OlesiaBarannik/HW5/3.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
+# Write a script that will calculate the factorial of the entered number without using recursion
+def factorial_number(user_number):
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW5/3.py:2]: good job
look at this code


def factorial_number(user_number):
    result = 1
    for i in range(1, user_number + 1):
        result *= i
    return result

print(factorial_number(3))

File: OlesiaBarannik/HW5/2.py (hunk 8 above)
Comment (OlesiaBarannik) [reply OlesiaBarannik/HW5/2.py:4]: thank you, I just thought that when 1 should be [0, 1, 1] that's why I used append(1)

File: OlesiaBarannik/HW5/3.py (hunk 9 above)
Comment (OlesiaBarannik) [reply OlesiaBarannik/HW5/3.py:2]: thank you) your variant is much better

File: OlesiaBarannik/HW6/1.py [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# Task1. In the range from 1 to 10 determine
+# • even numbers that are divisible by 2,
+# • odd numbers, which are divisible by 3,
+# • numbers that are not divisible by 2 and 3.
+
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW6/1.py:5]: good job

File: OlesiaBarannik/HW6/2.py [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# Task 2. Write a script that validates the login that the user enters.
+# If the login is "First", then welcome users. If there is a login
+# else, send an error message.
+# (need to use a while loop)
+
```
Comment (kolyasalubov) [inline OlesiaBarannik/HW6/2.py:5]: good job
look at this implementation

user_name = input('Hello, please input your Log in:')

while user_name != 'First':
    user_name = input('Error: wrong username, please try one more time. Username:')
else:
    print('Greeting. Access granted!!!', user_name)

**Pull request merged**